### DISCUSSION TITLE
Simple Icons

### DISCUSSION THREAD

**Discussion by is-ege** [tanujnotes/Olauncher#244] [General]
I really like this launcher but it is too text based, it's not really how our brain works.
I propose adding simplified and customizable icons for easily sorting out types of apps.
![test](https://user-images.githubusercontent.com/9782679/196474086-99b6bb2a-c9d8-42cf-98a3-a865687515ac.png)
Here is a quick proof of concept i made using https://feathericons.com

**Comment by tanujnotes** [comment]
It looks great. But once you add these minimal icons, things won't stop there. People will immediately start asking for icon pack support. I have absolutely no desire to deal with that.

**Reply by is-ege** [reply]
This is not intended to be an icon pack, just simple icons kinda to mentally group apps to be able to recognise them easier.
Same message bubble icon for every messaging app, same globe icon for every browser app, same note icon for every music app.
I too would disagree on the icon pack support, it definitely is not the aim of this project.

**Reply by is-ege** [reply]
Also I must add that it should be customizable, the user should be able to add any icon to any app they desire.

**Reply by tanujnotes** [reply]
I understand that it's not an icon pack but general users won't know that. They'll see this and be like - hey, if you can support one icon pack, I'm sure you can support more.

It's a nice idea but sorry, it won't be implemented. No app icons in Olauncher of any sorts.

What you can do, at least for the home screen apps, is that you can rename them to add emojis. Long press on home screen apps and start typing for RENAME option.

**Reply by is-ege** [reply]
Well you can simply deny their requests as you're doing right now. That seems to be your main concern with this feature.
People will always request unnecessary additions that go against the philosophy of said software. I think this would be a good addition that doesn't step over that line. But at the end of the day, you are the boss.

**Reply by tanujnotes** [reply]
Did you know that there's huge graveyard of great open source softwares which started with the best intentions, have tons of features and now completely abandoned. I don't want Olauncher to be one of them.

Features don't get added and work in isolation. They interact with other features, exponentially increasing complexity with time.

I'll give you an example. In last few days, 3-4 updates have been rejected by Play Store due to Accessibility permission required for double tap to lock. They have changed their policy and I have made the changes required but it is still getting rejected. I contacted their customer care and after a few back-and-forth, no solution. Currently I'm busy so I'll get back to this probably next month.

You can argue that that's a non-technical issue. Well, here's a technical one. I updated the libraries in Olauncher maybe first time in 2 years. That broke the default app drawer animation because that's Android for you. No problem. I added custom animation files. That broke the auto-open of keyboard. I fixed that and it's working fine on my phone but I'll only know for sure when the update get released.

That's just last 1-2 weeks.

I have said no to almost everything in past 2 years and it worked out great, both for me and thousands of people who love Olauncher, so I'll continue doing that. Besides, every hour I spend on Olauncher, I don't spend that hour working on my other projects. As you can see, it's more about priorities and less about "philosophy of said software".

I'm very happy with where Olauncher is today and my only goal is to keep it running as it is, for the longest time and with minimal effort. If you want the icons so bad, pay someone and they'll create a fork and add it for you. Beauty of open source. And next time you comment on any open source project, try to tone down the entitlement. Remember, they don't owe you anything.

I'd like this conversation to end now so I won't be commenting any further. Feel free to make the closing remarks if you wish.

**Reply by svet-am** [reply]
@tanujnotes - I understand (and largely agree) with your stance on the minimalism of OLauncher.  It's part of why I downloaded it form the store to begin with.  I'm running into issues where I have multiple applications with the same name in the app try - they are all various 2-factor authentication applications and they all show up as "Authenticator"  One is the Blizzard BattleNet authenticator, another is the Google authenticator, and the third is the Microsoft Authenticator.  My thinking is to simply show a very small version of the apps native icon right next to the text.  This would be very similar to how Windows Phone did things.  As it is right now, OLauncher is actually making me _less_ productive when I use it because it takes longer to figure out which application to open.

**Comment by is-ege** [comment]
I've never meant to assert any form of entitlement. Your priorities are to deliver a working product instead of a feature-packed bugfest. My only goal is to offer and defend my opinion which I think enhances the experience. If it came across as anything more than that I apologise. I am grateful for the time you spend on your projects and hope you keep up the great work.

**Comment by iSaluki** [comment]
One of the main selling points of Olauncher is the lack of icons, it really helps me stay distraction free, and I feel like adding any form of icons will lead to eventual feature creep. I see why people might want this, and even why it may end up being added, but I personally do not want this.

**Comment by Heike-HS** [comment]
I love and use Olauncher just because it is totally text based. Couldn't be more perfect for me.

**Comment by arcstur** [comment]
I think this can be easily solved!! Hold the name of the app in your home screen, a new screen to choose a new app will appear right? But! If you start typing a new name, it has the option to **rename** the app in the home screen. Then, just add an emoji from your keyboard :D

**Reply by Ucill1** [reply]
Thanks for this. I didn't notice that feature of renaming.
